Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+    bool passConvVeto = !ConversionTools::hasMatchedConversion(*(&el), conversions, theBeamSpot->position());
+      if (!passConvVeto) continue;
+      if (fabs(el.eta())>EtaTrack_Cut) continue;
+      if (el.pt()<PtEl_Cut) continue;
+      const Track * eltrack= el.bestTrack();
+      auto ElTrack=std::make_shared<reco::Track>(*eltrack);
+      ElTracks.push_back(ElTrack); object_container.push_back(nel);
+      el_pt.push_back(el.pt()); el_eta.push_back(el.eta()); el_phi.push_back(el.phi());
+      //     cout<<"el "<<nel<<" pt  "<<el.pt()<<endl;
+      nel++; object_id.push_back(11);
+    }
+//  cout<<nmuons<<" el "<<nel<<endl;
+  cleanedTracks.clear(); 
+  trk_index=0;
+  
+  for (const reco::Track& trk : *tracks){
+   if (!trk.quality(Track::highPurity)) continue;
+   if (trk.pt()<PtTrack_Cut) continue;
+   if (fabs(trk.eta())>EtaTrack_Cut) continue;
+   if(trk.charge()==0) continue;
+   if(trk.normalizedChi2()>MaxChi2Track_Cut || trk.normalizedChi2()<MinChi2Track_Cut) continue;
+   if (fabs(trk.dxy())/trk.dxyError()<TrackSdxy_Cut) continue;
+   double minDR=1000;
+   for (const reco::Muon& mu: *muons){
+      double tempDR=deltaR(mu.eta(),mu.phi(),trk.eta(),trk.phi());
+      if (minDR<tempDR) continue;
+      minDR=tempDR;
+   }
+   if (minDR<MuTrkMinDR_Cut) continue;
+   if (SelectedMu_DR<1000 ){
+     if (fabs(ZvertexTrg-trk.vz())>TrackMuDz_Cut ) continue;
+     if ( deltaR(trk.eta(),trk.phi(),SelectedTrgObj_PtEtaPhiCharge[1],SelectedTrgObj_PtEtaPhiCharge[2])<TrgExclusionCone) continue;
+   }
+   //assignments   
+   auto cleanTrack=std::make_shared<reco::Track>(trk);
+   cleanedTracks.push_back(cleanTrack);
+   Trk_container.push_back(trk_index);
+   trk_index++;
+ }
+ 
+    //create mother ee combination
+ 
+ // fit track pairs
+    cleanedObjTracks.clear(); cleanedPairTracks.clear();     
+    TLorentzVector vel1,vel2;
+    std::vector<std::shared_ptr<reco::Track>> cleanedObjects; 
+    //objects    
+    for(auto & vec: MuTracks) cleanedObjects.push_back(vec);
+    for(auto & vec: ElTracks) cleanedObjects.push_back(vec);  
+
+    if (cleanedObjects.size()==0)
+             return false;
+
+    for(auto& obj: cleanedObjects){
+      //      auto obj=cleanedObjects.at(iobj);
+      auto tranobj=std::make_shared<reco::TransientTrack>(reco::TransientTrack(*obj,&(*bFieldHandle)));
+      unsigned int iobj=&obj-&cleanedObjects[0];
+      unsigned int index=object_container.at(iobj);
+      if ( object_id.at(iobj)==13 && QualMu_Cut==1 && !muon_soft.at(index)) continue;
+     if ( object_id.at(iobj)==13 && QualMu_Cut==2 && !muon_medium.at(index)) continue;
+     if ( object_id.at(iobj)==13 && QualMu_Cut==3 && !muon_tight.at(index)) continue;
+     if (object_id.at(iobj)==13) vel1.SetPtEtaPhiM(muon_pt.at(index),muon_eta.at(index),muon_phi.at(index),0.0005);
+       
+     else vel1.SetPtEtaPhiM(el_pt.at(index),el_eta.at(index),el_phi.at(index),0.0005);
+    
+     
+      if (object_id.at(iobj)==13 && vel1.Pt()<PtMu_Cut) continue;
```

Comment:
wrong indentation

Reply:
this check could better be done as part of line 348

Reply:
Line 348 is inside another loop. If we place it there we are going to make some unneeded calculations